lib/addons/argocd/index.ts

```diff
@@ -161,11 +164,15 @@ export class ArgoCDAddOn implements spi.ClusterAddOn, spi.ClusterPostDeploy {
         const appRepo = this.options.bootstrapRepo;
 
         if (appRepo) {
+            // merge with custom bootstrapValues with executionContext and common values
```

Comment:
Can we check the ArgoGitOpsFactory setting and not perform this merge (which is going to be empty) if customers used approach 1  - app per add-on? Unless you have a use case for this as well.

Reply:
`ArgoGitOpsFactory` is overriding `KubectlProvider` static functions - I couldn't find a decent way to check the settings, although I can check the size of the `AddOnContexts` map:

```
let merged = this.options.bootstrapValues;
if (clusterInfo.getAddOnContexts().size > 0){
    const shared = { clusterName: clusterInfo.cluster.clusterName, region: Stack.of(clusterInfo.cluster).region };
    merged = { ...shared, ...Object.fromEntries(clusterInfo.getAddOnContexts().entries()), ...this.options.bootstrapValues };
}
```

I don't see the value adding this `if` though, because it will just merge an empty map `AddOnContexts` should the customer not choosing the `app of apps` mode.. or maybe the concern is merging the `shared` value?
@shapirov103 thoughts?

The other option is to add a property to `clusterInfo`, such as `argoGitOpsFactory` and:
- set the value to `enableGitOps` when `createArgoHelmApplication()` is executed.
- set the value to `enableGitOpsAppOfApps` when `generateArgoHelmApplicationValues` is executed.
It is not really a good approach either.